library/core/registry/src/main/java/org/quiltmc/qsl/registry/mixin/DynamicRegistrySyncMixin.java

```diff
@@ -17,21 +18,52 @@
 package org.quiltmc.qsl.registry.mixin;
 
 import java.util.Map;
+import java.util.stream.Stream;
 
+import org.spongepowered.asm.mixin.Dynamic;
 import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Mutable;
 import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
 
+import net.minecraft.registry.DynamicRegistryManager;
 import net.minecraft.registry.DynamicRegistrySync;
 import net.minecraft.registry.Registry;
 import net.minecraft.registry.RegistryKey;
 
+import org.quiltmc.qsl.registry.impl.dynamic.DynamicRegistryFlagManager;
+
 @Mixin(DynamicRegistrySync.class)
 public abstract class DynamicRegistrySyncMixin {
 	@SuppressWarnings("unused")	// makes the field mutable for use by the accessor
 	@Shadow
 	@Final
 	@Mutable
 	private static Map<RegistryKey<? extends Registry<?>>, ?> SYNCED_CODECS;
+	@Unique
+	private static boolean filterRegistryEntry(DynamicRegistryManager.RegistryEntry<?> entry) {
+		// OPTIONAL
+		if (DynamicRegistryFlagManager.isOptional(entry.key().getValue())) {
+			return entry.value().size() > 0;
+		}
+		return true; //if no flags apply, always return true
```

Comment:
hm, am not sure what are the general feelings about inline comments, but maybe it's fine here, with just an extra space :p
```suggestion
		}

		return true; // If no flags apply, always return true
```